### PR TITLE
Only use ipv4 addresses for the resolv.conf override

### PR DESCRIPTION
**Pull request by josecv**
Signed-off-by: Jose Cortes <josecortes@datawire.io>

## Description

A few sentences describing the overall goals of the pull request's commits.

## Checklist

 - [x] I made sure to update `./CHANGELOG.md`.
 - [ ] I made sure to add any docs changes required for my change (including release notes).
 - [ ] My change is adequately tested.
 - [ ] I updated `DEVELOPING.md` with any any special dev tricks I had to use to work on this code efficiently.
 - [ ] I updated `TELEMETRY.md` if I added, changed, or removed a metric name.
 - [ ] Once my PR is ready to have integration tests ran, I posted the PR in #telepresence-dev in the datawire-oss slack so that the "ok to test" label can be applied.

### CODE REVIEW

**Review by maxlegault — APPROVED** [review]

File: pkg/client/rootd/dns/server_linux.go [hunk 1]
```diff
@@ -105,8 +105,12 @@ func (s *Server) runOverridingServer(c context.Context, dev *vif.Device) error {
 		for _, line := range strings.Split(string(dat), "\n") {
 			if strings.HasPrefix(strings.TrimSpace(line), "nameserver") {
 				fields := strings.Fields(line)
-				s.config.LocalIp = net.ParseIP(fields[1])
-				dlog.Infof(c, "Automatically set -dns=%s", net.IP(s.config.LocalIp))
+				ip := net.ParseIP(fields[1])
+				if ip.To4() != nil {
+					s.config.LocalIp = ip.To4()
+					dlog.Infof(c, "Automatically set -dns=%s", net.IP(s.config.LocalIp))
+					break
```
Comment (thallgren) [inline pkg/client/rootd/dns/server_linux.go]: This break is wrong. It will prevent parsing of "search" if that comes after "nameserver".

**Review by thallgren — CHANGES_REQUESTED** [review]

File: pkg/client/rootd/dns/server_linux.go (hunk 1 above)
Comment (thallgren) [reply pkg/client/rootd/dns/server_linux.go]: What would it take to actually support IPv6?

Comment (josecv) [reply pkg/client/rootd/dns/server_linux.go]: ah ok good catch thank you

Comment (josecv) [reply pkg/client/rootd/dns/server_linux.go]: > What would it take to actually support IPv6?

not sure really i think there might be a way to make iptables work with it (i believe it's a matter of using iptables6 but i've never done it), but rn it fails

**Review by thallgren — APPROVED** [review]

File: pkg/client/rootd/dns/server_linux.go (hunk 1 above)
Comment (thallgren) [reply pkg/client/rootd/dns/server_linux.go]: Not for this release, for sure.

**Comment by thallgren** [comment]
Merged

**Pull request closed without merging**